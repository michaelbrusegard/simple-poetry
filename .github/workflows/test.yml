name: Test

on:
  push:
    branches:
      - main

jobs:
  test:
    name: Run Tests
    runs-on: self-hosted
    steps:
      - name: Checkout code
        uses: actions/checkout@v4

      - name: Set up Node.js
        uses: actions/setup-node@v4
        with:
          node-version: '22'

      - name: Install dependencies
        run: npm install

      - name: Run tests
        run: npx vitest run
