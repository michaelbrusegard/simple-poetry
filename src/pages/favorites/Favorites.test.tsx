import { screen } from '@testing-library/react'
import { render } from '../../test/utils'
import FavoritesPage from './index'

describe('FavoritesPage tests', () => {
  it('should render the h1 element with the correct text', () => {
    render(<FavoritesPage />)
    expect(screen.getByText('Favorites')).toBeInTheDocument()
  })

  it('should render the link component correctly', () => {
    render(<FavoritesPage />)
    expect(screen.getByText('Home')).toHaveAttribute('href', '/project1/')
  })

  it('should match the snapshot', async () => {
    const { asFragment } = render(<FavoritesPage />)
    const fragment = asFragment()
    expect(fragment).toMatchSnapshot()
  })
})
