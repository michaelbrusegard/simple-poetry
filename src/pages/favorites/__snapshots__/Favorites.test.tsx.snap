// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FavoritesPage tests > should match the snapshot 1`] = `<DocumentFragment />`;
